### ISSUE TITLE
bench-tps fails to discover local bootstrap validator

### ISSUE THREAD

**Issue by apfitzge** [solana-labs/solana#30157]
#### Problem
bench-tps failing to discover my bootstrap validator.

Steps to reproduce:
Terminal 1: `$ NDEBUG=1 ./multinode-demo/setup.sh && NDEBUG=1 ./multinode-demo/faucet.sh`
Terminal 2: `$ NDEBUG=1 ./multinode-demo/bootstrap-validator.sh`
Terminal 3: `$ NDEBUG=1 ./multinode-demo/bench-tps.sh`

#### Proposed Solution
git bisect and figure out where this got broken

**Comment by apfitzge** [comment]
```
b29a50a55e7802a7b1deb9d4a66bac704d349a87 is the first bad commit
commit b29a50a55e7802a7b1deb9d4a66bac704d349a87
Author: Brennan <brennan.watt@solana.com>
Date:   Wed Feb 1 11:16:15 2023 -0800

    Exclude loopback from valid IP addresses (#29992)
    
    Excludes loopback IP addresses from being valid IP addresses unless private addresses are explicitly allowed

 ci/localnet-sanity.sh         |  5 ++++-
 docs/src/cluster/bench-tps.md |  7 +++++++
 streamer/src/socket.rs        | 13 ++++++-------
 3 files changed, 17 insertions(+), 8 deletions(-)
```

@bw-solana - No need to look immediately, but FYI this seems to have broken  bench-tps runs locally (which I do a lot). Seems clear why from the commit message. I'll look into the reason for this commit, and then I'm hoping bench-tps or one of the scripts I'm running takes an easy default command.

**Comment by bw-solana** [comment]
Are you able to pass allow-private-addr and see the validators show up?

**Comment by apfitzge** [comment]
> Are you able to pass `--allow-private-addr` and see the validators show up? I'm hoping passing this into bootstrap-validator should resolve the issue for you. If not, I'll jump on it

Passing it in worked - made a PR to just add that to the default args of the script.

**Comment by apfitzge** [comment]
Closing will not fix - see discussion on #30163 
Just pass `--allow-private-addr` to `bootstrap-validator.sh` from now on if you need to run locally.
